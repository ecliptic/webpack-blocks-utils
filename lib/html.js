'use strict';

exports.__esModule = true;
exports.html = html;

var _htmlWebpackPlugin = require('html-webpack-plugin');

var _htmlWebpackPlugin2 = _interopRequireDefault(_htmlWebpackPlugin);

var _deepmerge = require('deepmerge');

var _deepmerge2 = _interopRequireDefault(_deepmerge);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Webpack block for html-webpack-plugin
 *
 * @see https://github.com/ampedandwired/html-webpack-plugin
 */
function html(options) {
  return Object.assign(function (context) {
    return function (prevConfig) {
      var defaultOpts = {
        filename: 'index.html',
        template: 'templates/index.html',
        showErrors: false

        // Merge the provided html config into the context
      };var html = context.html || defaultOpts;

      context.html = (0, _deepmerge2.default)(html, options, { clone: true }

      // Don't modify the config yet
      );return prevConfig;
    };
  }, { post: postConfig });
}

function postConfig(context, _ref) {
  var merge = _ref.merge;

  var plugin = new _htmlWebpackPlugin2.default(context.html);
  return merge({ plugins: [plugin] });
}