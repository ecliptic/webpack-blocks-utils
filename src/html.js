// @flow
/**
 * Webpack block for html-webpack-plugin
 *
 * @see https://github.com/ampedandwired/html-webpack-plugin
 */
import HtmlWebpackPlugin from 'html-webpack-plugin'
import deepmerge from 'deepmerge'

import type {Block, Helpers, Config} from './types'

export function html (options): Block {
  return Object.assign(
    context => prevConfig => {
      const defaultOpts = {
        filename: 'index.html',
        template: 'templates/index.html',
        showErrors: false,
      }

      // Merge the provided html config into the context
      const html = context.html || defaultOpts

      context.html = deepmerge(html, options, {clone: true})

      // Don't modify the config yet
      return prevConfig
    },
    {post: postConfig}
  )
}

function postConfig (context: Object, {merge}: Helpers): Config => Config {
  const plugin = new HtmlWebpackPlugin(context.html)
  return merge({plugins: [plugin]})
}
