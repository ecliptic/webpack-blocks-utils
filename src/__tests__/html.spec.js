// @flow
import {html} from '../html'

describe('html()', () => {
  test('single definition', () => {
    const block = html({template: 'assets/index.html'})
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const intermediateConfig = block(context, helpers)({})
    if (block.post) block.post(context, helpers)(intermediateConfig)

    expect(context).toMatchSnapshot()
  })

  test('multiple definitions', () => {
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    html({showErrors: true})(context, helpers)({})

    // This should be overridden by the below definition
    html({template: 'assets/home.html'})(context, helpers)({})

    const block = html({template: 'templates/index.html'})
    const intermediateConfig = block(context, helpers)({})
    if (block.post) block.post(context, helpers)(intermediateConfig)

    expect(context).toMatchSnapshot()
  })
})
