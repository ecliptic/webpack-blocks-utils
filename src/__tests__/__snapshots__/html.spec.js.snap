// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html() multiple definitions 1`] = `
Object {
  "html": Object {
    "filename": "index.html",
    "showErrors": true,
    "template": "templates/index.html",
  },
}
`;

exports[`html() single definition 1`] = `
Object {
  "html": Object {
    "filename": "index.html",
    "showErrors": false,
    "template": "assets/index.html",
  },
}
`;
